src/main/java/uk/nhs/digital/nhsconnect/nhais/service/edifact_to_fhir/FP69PriorNotificationTransactionMapper.java

```diff
@@ -0,0 +1,193 @@
+            .ifPresentOrElse(
+                patient::setBirthDate,
+                () -> {
+                    throw new EdifactValidationException(
+                        "For an FP69 prior notification (reference F9) the DTM+329 segment is required to provide the patient date of birth");
+                });
+    }
+
+    private void mapAddress(Patient patient, Transaction transaction) {
+        transaction.getPersonAddress()
+            .ifPresent(personAddress -> {
+                var address = patient.addAddress();
+                mapPostalCode(personAddress, address);
+                mapAddressLines(personAddress, address);
+            });
+    }
+
+    private void mapReasonCode(Parameters parameters, Transaction transaction) {
+        transaction.getFp69ReasonCode()
+            .map(FP69ReasonCode::getCode)
+            .map(Object::toString)
+            .map(StringType::new)
+            .map(reasonCode -> new Parameters.ParametersParameterComponent()
+                .setName(ParameterNames.FP69_REASON_CODE)
+                .setValue(reasonCode))
+            .ifPresentOrElse(
+                parameters::addParameter,
+                () -> {
+                    throw new EdifactValidationException(
+                        "For an FP69 prior notification (reference F9) the HEA+FRN segment is required");
+                });
+    }
+
+    private void mapExpiryDate(Parameters parameters, Transaction transaction) {
+        transaction.getFp69ExpiryDate()
+            .map(FP69ExpiryDate::getExpiryDate)
+            .map(expiryDate -> expiryDate.format(DATE_FORMAT))
+            .map(StringType::new)
+            .map(expiryDate -> new Parameters.ParametersParameterComponent()
+                .setName(ParameterNames.FP69_EXPIRY_DATE)
+                .setValue(expiryDate))
+            .ifPresentOrElse(
+                parameters::addParameter,
+                () -> {
+                    throw new EdifactValidationException(
+                        "For an FP69 prior notification (reference F9) the DTM+962 segment is required");
+                });
+    }
+
+    private void mapFreeText(Parameters parameters, Transaction transaction) {
+        transaction.getFreeText()
+            .map(FreeText::getTextLiteral)
+            .map(StringType::new)
+            .map(text -> new Parameters.ParametersParameterComponent()
+                .setName(ParameterNames.FREE_TEXT)
+                .setValue(text))
+            .ifPresent(parameters::addParameter);
+    }
+
+    private void mapAddressLines(PersonAddress personAddress, Address address) {
+        address.setLine(Stream.of(
+            Optional.ofNullable(personAddress.getAddressLine1()),
+            Optional.ofNullable(personAddress.getAddressLine2()),
+            Optional.ofNullable(personAddress.getAddressLine3()),
+            Optional.ofNullable(personAddress.getAddressLine4()),
+            Optional.ofNullable(personAddress.getAddressLine5()))
```

Comment:
These Optionals are redundant

Reply:
Ok, I was kind of wrong as `Stream.of()` doesn't support nulls, but... that would work:
````java
address.setLine(Stream.builder()
	.add(personAddress.getAddressLine1())
	.add(personAddress.getAddressLine2())
	.add(personAddress.getAddressLine3())
	.add(personAddress.getAddressLine4())
	.add(personAddress.getAddressLine5())
	.build()
	.map(NullableStringType::new)
    .map(StringType.class::cast)
    .collect(Collectors.toList()));
````